#  Статья-исследование "Развитие игровой индустрии в начале XXI века"

- Автор: Дмитрий Ниткин
- Дата: 03.03.2026

# Цели и задачи проекта

- Привлечение внимание людей, которые любят старые игры, и заодно заинтересовать их «Секретами Темнолесья»;
- Комплексный анализ развития игровой индустрии в период 2000-2013 гг;
- Анализ объёмов продаж игр разных жанров и выпущенных на разных платформах, а также пользовательские и экспертные оценки игр.

# Описание данных

Данные `/datasets/new_games.csv` содержат информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр:

- `Name` — название игры;
- `Platform` — название платформы;
- `Year of Release` — год выпуска игры;
- `Genre` — жанр игры;
- `NA sales` — продажи в Северной Америке (в миллионах проданных копий);
- `EU sales` — продажи в Европе (в миллионах проданных копий);
- `JP sales` — продажи в Японии (в миллионах проданных копий);
- `Other sales` — продажи в других странах (в миллионах проданных копий);
- `Critic Score` — оценка критиков (от 0 до 100);
- `User Score` — оценка пользователей (от 0 до 10);
- `Rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.

# Содержимое проекта:

## Загрузка данных и знакомство с ними

- Загрузите необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`.


In [1]:
# Подключаем библиотеку pandas
import pandas as pd

In [2]:
# Выгружаем данные из датасета hotel_dataset.csv в датафрейм games
games = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')

- Познакомьтесь с данными: выведите первые строки и результат метода `info()`.


In [3]:
# Выводим информацию о датафрейме
display(games.info())

# Сохраним объем датафрейма в отдельной переменной для сравления после удаления дубликатов и пустых строк
total_rows = games.shape[0]
display(f'Кол-во строк датафрейма до предобработкаи данных {total_rows}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


None

'Кол-во строк датафрейма до предобработкаи данных 16956'

**Информация о данных:**

1. Датафрейм содержит 11 столбцов
2. Датафрем содержит 16 956 строчек
3. Данные в датафреме двух типов: object и float64
4. Столбец Critic Score - содержит наибольшее количество пропусков

In [4]:
# Выводим первые 5 и последние 5 строк датафрейма на экран
display(games.head(5))
display(games.tail(5))

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN


,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
16951,Samurai Warriors: Sanada Maru,PS3,2016.0,Action,0.00,0.0,0.01,0.0,NaN,NaN,NaN
16952,LMA Manager 2007,X360,2006.0,Sports,0.00,0.01,0.0,0.0,NaN,NaN,NaN
16953,Haitaka no Psychedelica,PSV,2016.0,Adventure,0.00,0.0,0.01,0.0,NaN,NaN,NaN
16954,Spirits & Spells,GBA,2003.0,Platform,0.01,0.0,0.0,0.0,NaN,NaN,NaN
16955,Winning Post 8 2016,PSV,2016.0,Simulation,0.00,0.0,0.01,0.0,NaN,NaN,NaN


In [5]:
# Определяем период, за который присутствуют данные в датафрейме
min_date = games['Year of Release'].min()
max_date = games['Year of Release'].max()
f"Минимум - {min_date}. Максимум - {max_date}"

'Минимум - 1980.0. Максимум - 2016.0'

**В датафрейме нахоятся данные с 1980 года по 2016 год.**

In [6]:
# Выборочная проверка уникальных значений
games['Genre'].unique()

array(['Sports', 'Platform', 'Racing', 'Role-Playing', 'Puzzle', 'Misc',
       'Shooter', 'Simulation', 'Action', 'Fighting', 'Adventure',
       'Strategy', nan, 'MISC', 'ROLE-PLAYING', 'RACING', 'ACTION',
       'SHOOTER', 'FIGHTING', 'SPORTS', 'PLATFORM', 'ADVENTURE',
       'SIMULATION', 'PUZZLE', 'STRATEGY'], dtype=object)

**В датафрейме присутствуют, как явные, так и неявные дубликаты. Пример жанры - Racing, RACING.**

---

## Проверка ошибок в данных и их предобработка


### Названия, или метки, столбцов датафрейма

- Выведите на экран названия всех столбцов датафрейма и проверьте их стиль написания.
- Приведите все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [7]:
games.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [8]:
# Переименовываем названия колонок в snake case
games.columns = games.columns.str.lower().str.replace(" ", "_")

In [9]:
games.columns

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='object')

### Типы данных

- Если встречаются некорректные типы данных, предположите их причины.
- При необходимости проведите преобразование типов данных. Помните, что столбцы с числовыми данными и пропусками нельзя преобразовать к типу `int64`. Сначала вам понадобится обработать пропуски, а затем преобразовать типы данных.

**Числовые значения с плавающей точкой (float64).**
- К колонкам `NA sales`, `Other sales`, `Critic Score` применен правильный тип данных, но можно уменьшить разрядность до 32.
- Для колонки `Year of Release` тип `float64` некорректен. Для работы со временем обычно происходят преобразования в тип данных `datetime64`, который позволит корректно производить работу с данными. Т.к нам необходим только год, то при вычислении через `.dt.year` присвоится тип данных отличный от даты. Можно преобразовать в тип данных `integer` после обработки NaN.

**Строковые данные (object).**
- Колонка `Name` содержит строковую информацию, к ней применен корректный тип данных.
- Колонки `Platform`, `Genre`, `Rating` также хранят текстовые данные, но их можно рассматривать как категориальные признаки. Для них можно использовать тип `category` для оптимизации памяти и улучшении производительности.
- Колонки `EU sales`, `JP sales`, `User Score` содержат числовые значения с плавающей точкой. Необходимо преобразовать данные колонки в тип данных `float32` для произведения расчетов.

In [10]:
# Меняем тип данных в category
columns_name = ['platform', 'genre', 'rating']
for column in columns_name:
    games[column] = games[column].astype('category')

- В числовых столбцах могут встретиться строковые значения, например `unknown` или другие. Приводите такие столбцы к числовому типу данных, заменив строковые значения на пропуски.

In [11]:
# Уменьшаем разрядность и преобразуем колонки через pd.to_numeric для обработки ошибок и автоматического выбора разрядности
columns_name = ['na_sales', 'eu_sales', 'jp_sales', 'other_sales', 'critic_score', 'user_score']
for column in columns_name:
    games[column] = pd.to_numeric(games[column], errors='coerce', downcast='float')

###  Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [12]:
# Считаем абсолютные кол-ва пропусков
games.isna().sum()

name                  2
platform              0
year_of_release     275
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

In [13]:
# Считаем относительные кол-ва пропусков
round(games.isna().mean()*100, 2)

name                0.01
platform            0.00
year_of_release     1.62
genre               0.01
na_sales            0.00
eu_sales            0.04
jp_sales            0.02
other_sales         0.00
critic_score       51.39
user_score         54.66
rating             40.52
dtype: float64

- Изучите данные с пропущенными значениями. Напишите промежуточный вывод: для каких столбцов характерны пропуски и сколько их. Предположите, почему пропуски могли возникнуть. Укажите, какие действия с этими данными можно сделать и почему.


**Анализ пропусков в значениях:**

- 8 колонок содержат пропуска.
- Пропуски в колонках `eu_sales` и `jp_sales` допустимы и могут быть связаны остутствием продаж или отсутствием информации о продажах. Т.К. процент этих пропусков мал **0.04%** и **0.02%**, в рассчетах мы можем обозначить их как среднее значение по платформе и дате выхода игры. Такие пропуска относятся к типу MAR или MNAR. Можно изучить их распределение и понять к какому типу их отнести.
- Пропуски в колонках `critic_score` и `user_score` могут обозначать отсутствие оценок игры или отсутствие данных об оценках игры. Эти пропуски занимают **более 50%** всех данных. Эти пропуска необходимо обработать, т.к. по ним надо делать категорирование данных. Такие пропуска относятся к типу *MAR*. Для игр могут отсутствовать оценки по причинам возраста игр, неактуальности платформы в год выпуска игры и региональных осбенностей. Необходимо изучить их детальнее и принять решение по их обработке.
- Колонка `rating` содержит **40%** значений NaN. По данным из интернета, рейтинг может отсутствовать по множеству причин (отсутствие денег у маленьких студий, региональные системы рейтингов и т.п.). Эта колонка не нужна для расчетов, поэтому можем их оставить. Такие пропуска относятся к типу *MNAR*.
- Колонки `name` и `genre` содержат **0.01%** пропусков. Это может быть связано с технической ошибкой занесения данных. Можем не учитывать их в расчетах. Такие пропуска относятся к типу *MCAR*.
- Колонка `year_of_release` содержит **1.62%** пропусков. Это может быть связано с технической ошибкой занесения данных или отсутсвие релиза игры. Значения выхода игр можно собрать вручную для каждой игры при необходимости, но это будет ресурсозатратно. Т.к. процент NaN мал, можем не учитывать их для расчетов. Такие пропуска относятся к типу *MNAR* или *MCAR*.

- Обработайте пропущенные значения. Для каждого случая вы можете выбрать оптимальный, на ваш взгляд, вариант: заменить на определённое значение, оставить как есть или удалить.
- Если вы решите заменить пропуски на значение-индикатор, то убедитесь, что предложенное значение не может быть использовано в данных.
- Если вы нашли пропуски в данных с количеством проданных копий игры в том или ином регионе, их можно заменить на среднее значение в зависимости от названия платформы и года выхода игры.

In [14]:
# Удаляем NaN для колонок name, genre
new_games = games.dropna(subset=['name', 'genre'])

In [15]:
# Убираем все строки с Nan в дате
new_games = new_games.dropna(subset=['year_of_release'])

In [16]:
# Замена пропусков на средние значения в колонках eu_sales и jp_sales
def mean_group_sales(row, column):
    if pd.isna(row[column]):
        group = new_games[(new_games['platform'] == row['platform']) & 
                             (new_games['year_of_release'] == row['year_of_release'])]
        return group[column].mean()
    else:
        return row[column]
    
columns = ['eu_sales', 'jp_sales']
for column in columns:
    new_games[column] = new_games.apply(lambda row: mean_group_sales(row, column), axis=1)

In [17]:
# Замена значений NaN в 'critic_score' на -1
new_games['critic_score'] = new_games['critic_score'].fillna(-1)

In [18]:
# Замена значений NaN в 'user_score' на -1
new_games['user_score'] = new_games['user_score'].fillna(-1)

In [19]:
# Анализ NaN значений после обработки
new_games[new_games['year_of_release'].between(2000, 2013)].isna().sum()

name                  0
platform              0
year_of_release       0
genre                 0
na_sales              0
eu_sales              0
jp_sales              0
other_sales           0
critic_score          0
user_score            0
rating             4133
dtype: int64

### Явные и неявные дубликаты в данных

- Изучите уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
- При необходимости проведите нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

In [20]:
# Проверка неявных дубликатов в колонке platform
new_games.sort_values('platform')['platform'].unique().to_list()

['2600',
 '3DO',
 '3DS',
 'DC',
 'DS',
 'GB',
 'GBA',
 'GC',
 'GEN',
 'GG',
 'N64',
 'NES',
 'NG',
 'PC',
 'PCFX',
 'PS',
 'PS2',
 'PS3',
 'PS4',
 'PSP',
 'PSV',
 'SAT',
 'SCD',
 'SNES',
 'TG16',
 'WS',
 'Wii',
 'WiiU',
 'X360',
 'XB',
 'XOne']

Неявные дубликаты в данных отсутствуют

In [21]:
# Проверка неявных дубликатов в колонке genre
new_games.sort_values('genre')['genre'].unique().to_list()

['ACTION',
 'ADVENTURE',
 'Action',
 'Adventure',
 'FIGHTING',
 'Fighting',
 'MISC',
 'Misc',
 'PLATFORM',
 'PUZZLE',
 'Platform',
 'Puzzle',
 'RACING',
 'ROLE-PLAYING',
 'Racing',
 'Role-Playing',
 'SHOOTER',
 'SIMULATION',
 'SPORTS',
 'STRATEGY',
 'Shooter',
 'Simulation',
 'Sports',
 'Strategy']

Неявные дубликаты в данной колонке присутствуют и связаны с различным регистром.

In [22]:
# Преобразование колонки 'genre' в нижний регистр
new_games['genre'] = new_games['genre'].str.lower()

In [23]:
# Проверка неявных дубликатов в колонке rating
new_games.sort_values('rating')['rating'].unique()

['AO', 'E', 'E10+', 'EC', 'K-A', 'M', 'RP', 'T', NaN]
Categories (8, object): ['AO', 'E', 'E10+', 'EC', 'K-A', 'M', 'RP', 'T']

Неявные дубликаты в данной колонке присутствуют. В рейтинге указаны региональная категория K-A,

In [24]:
# Замена рейтинга согласно стандарту рейтингов ESRB
new_games['rating'] = new_games['rating'].replace('K-A', 'E')

In [25]:
new_games['name'].shape[0]

16679

In [26]:
# Преобразование колонки 'name' в нижний регистр для поиска дубликатов
new_games['name'] = new_games['name'].str.lower()

- После того как нормализуете данные и устраните неявные дубликаты, проверьте наличие явных дубликатов в данных.

In [27]:
# Найдем кол-во явных дубликатов
new_games.duplicated().sum()

235

In [28]:
# Удалим явные дубликаты
new_games = new_games.drop_duplicates()

In [29]:
# Найдем неявные дубликаты по ряду колонок после очистки датафрема
new_games_sorted = new_games.sort_values(by=new_games.columns.tolist())
new_games_sorted[new_games_sorted.duplicated(subset=['name', 'platform', 'year_of_release', 'genre'],keep=False)]

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
16465,madden nfl 13,PS3,2012.0,sports,0.00,0.01,0.0,0.00,83.0,5.5,E
606,madden nfl 13,PS3,2012.0,sports,2.11,0.22,0.0,0.23,83.0,5.5,E


In [30]:
# Объединим данное значение вручную, чтобы не нагружать сервер ради одного вычисления
new_games.loc[606, 'eu_sales'] += 0.01
new_games = new_games.drop_duplicates(subset=['name', 'platform', 'year_of_release', 'genre'])

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

In [31]:
new_df_len = new_games.shape[0]
old_df_len = games.shape[0]
new_df_len

16443

In [32]:
old_df_len - new_df_len

513

In [33]:
(1-new_df_len/old_df_len)*100

3.0254777070063743

- Всего в новом датафрейме **16443** строки
- При обработке датафрейма были удалены **513** строк
- Доля удаленных строк составляет **3%**

- После проведения предобработки данных напишите общий промежуточный вывод.

**Нулевые значения:**
- В столбцах user_score и critic_score присутствовало множество дубликатов. Эти данные необходимы нам для группировки согласно ТЗ. Отсутствие рейтинга, скорее всего, не связано с технической ошибкой, поэтому было принято решение заменить данные без оценки на индикатор -1.
- Нулевые значения из столбцов с продажами были обработаны и сохранены как среднее значение по платформе и году релиза.
- Нулевые значения в названии игры, жанре и годе выпуска незначительны и были удалены.

**Дубликаты:**
- Неявные дубликаты в данных были связаны с различным регистром
- Неявные дубликаты в данных были удалены

---

## Фильтрация данных

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [34]:
df_actual = new_games[new_games['year_of_release'].between(2000, 2013)]

In [35]:
df_actual.shape[0]

12780

- Отобраны данные с **2000 по 2013** год включительно
- За эти даты в датафрейме содержатся **12780** записей. **77%** процентов всех записей содержатся в этом временном периоде.

---

## Категоризация данных
    
Проведите категоризацию данных:
- Разделите все игры по оценкам пользователей и выделите такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [36]:
# Создаем копию датафрейма. Без копии выдается ошибка
df_actual = df_actual.copy()

In [37]:
# Создаем категории по колонке user_score
df_bins = [-1, 0, 3, 8, 11]
df_lables = ['Нет оценки', 'Низкая оценка', 'Средняя оценка', 'Высокая оценка']
df_actual['user_score_category'] = pd.cut(df_actual['user_score'], bins=df_bins, labels=df_lables, right=False)

- Разделите все игры по оценкам критиков и выделите такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [38]:
# Создаем категории по колонке user_score
df_bins = [-1, 0, 30, 80, 101]
df_lables = ['Нет оценки', 'Низкая оценка', 'Средняя оценка', 'Высокая оценка']
df_actual['critic_score_category'] = pd.cut(df_actual['critic_score'], bins=df_bins, labels=df_lables, right=False)

Все рейтинги категорированы на `Нет оценки`, `Низкая оценка`, `Средняя оценка`, `Высокая оценка`.

- После категоризации данных проверьте результат: сгруппируйте данные по выделенным категориям и посчитайте количество игр в каждой категории.

In [39]:
# Считаем общую сумму продаж
df_actual['total_sales'] = df_actual['na_sales'] + df_actual['eu_sales'] + df_actual['jp_sales'] + df_actual['other_sales']

In [40]:
# Создаем функцию для группировки
def result_category(column_name):
    result = df_actual.groupby(column_name).agg({'name': 'count','total_sales': 'sum'})
    result = result.sort_values(by='name', ascending=False).reset_index()
    result['count_share'] = round(result['name']/ result['name'].sum() *100,1)
    result = result.rename(columns={'name': 'count'})
    result['total_sales'] = round(result['total_sales'], 1)
    result['avg_sales_per_game'] = round((result['total_sales'] / result['count']) * 1000)
    return result

In [41]:
# Отображаем категории оценки пользователей
result_category('user_score_category')

,user_score_category,count,total_sales,count_share,avg_sales_per_game
0,Нет оценки,6298,1569.2,49.3,249.0
1,Средняя оценка,4080,2622.3,31.9,643.0
2,Высокая оценка,2286,2185.7,17.9,956.0
3,Низкая оценка,116,50.0,0.9,431.0


- **49**% игр не получили оценок. Это `6298` проектов.
- **32**% игр попали в категорию со средними оценками. Это `4080` проектов - `2622.3 млн.` проданных копий
- **18**% игр попали в категорию с высокой оценкой (оценка от 8 до 10). Это `2286` игр - `2185.7 млн.` проданных копий
- **1**% всех игр попал в категорию с низкой оценкой
- В среднем продается по **249 тыс** копий на игры `без оценок`, **643 тыс** копий на игры со `средним рейтигом`, **956 тыс** копий на игры с `высоким рейтингом` и **431 тыс** копий на игры с `низкими оценками`.

In [42]:
# Отображаем категории оценки критиков
result_category('critic_score_category')

,critic_score_category,count,total_sales,count_share,avg_sales_per_game
0,Нет оценки,5612,1511.4,43.9,269.0
1,Средняя оценка,5422,2362.1,42.4,436.0
2,Высокая оценка,1691,2541.4,13.2,1503.0
3,Низкая оценка,55,12.3,0.4,224.0


- **44**% игр не получили оценок от критиков. Это `5612` проектов.
- **42**% игр попали в категорию со средними оценками. Это `5422` проектов - `2362.1` млн. проданных копий
- **13**% игр попали в категорию с высокой оценкой (оценка от 8 до 10). Это `1691` игр - `2541.4` млн. проданных копий
- **0.4**% всех игр попал в категорию с низкой оценкой
- В среднем продается по **269 тыс** копий на один проект `без оценок`, **436 тыс** копий на игры со `средним рейтигом`, **1.5 млн** копий на игры с `высоким рейтингом` и **224 тыс** копий на игры с `низкими оценками`.

- Выделите топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [43]:
# Выделяем топ-7 платформ по кол-ву игр за период с 2000 по 2013 включительно
result = df_actual.groupby('platform').agg({'name': 'count','total_sales': 'sum'})
result = result.sort_values(by='name', ascending=False).reset_index()
result = result.rename(columns={'name': 'count'})
result['total_sales'] = round(result['total_sales'], 1)
result.head(7)

,platform,count,total_sales
0,PS2,2127,1232.9
1,DS,2120,801.9
2,Wii,1275,886.1
3,PSP,1180,289.2
4,X360,1121,912.5
5,PS3,1086,863.2
6,GBA,811,312.5


Топ 7 платформ по кол-ву выпущенных игр.

- **На первом месте** - PS2 - Sony Playstation 2 (`2127` игры, `1233` млн. проданных копий)
- **На седьмом** - GBA - Game Boy Advance (`811` проектов, `312` млн. проданных копий)

---

## Итоговый вывод

В конце напишите основной вывод и отразите, какую работу проделали. Не забудьте указать описание среза данных и новых полей, которые добавили в исходный датасет.

Были загружены данные `new_games.csv`. Они содержат 11 колонок и 16 956 строк, в которых представлена информация об играх, их жанрах, годе выпуска, продажах в различных регионах и рейтингах. При первичном знакомстве с данными и их предобработкой получены результаты:

- Восемь столбцов (`year_of_release`, `eu_sales`, `jp_sales`, `critic_score`, `user_score`, `name`, `genre`, `rating`). Максимальное значение пропущенных данных в столбце `critic_score` - 51.39%.

- Для работы с пропусками были сделаны следующие действия:

    - `year_of_release`. NaN заменены на индикатор -1 и не учтены в расчетах
    - `eu_sales, jp_sales`. NaN расчитаны, как среднее значение по платформе и дате выхода игры
    - `name`, `genre`. NaN значения составляли менее 2%, было принято решение не учитывать их в расчетах
    - `rating`. NaN значения были оставлены, т.к. нам не нужен данный рейтинг для расчетов
    - `critic_score`, `user_score`. NaN заменены на индикатор -1. В группировке данные значения попали в категорию игр без оценок.

- Для оптимизации работы с данными в датафрейме были произведены следующие изменения типов данных:

    - Для колонок `platform`, `genre`, `rating` тип данных изменен с `object` на `category`
    - Для колонок `EU sales`, `JP sales`, `User Score` проведено преобразование в `float32` для произведения расчетов

- Для дополнительной работы с данными был создан новый датафрейм `games_df_actual`:

    - Сделана фильтрация по году выпуска игры (с 2000 по 2013 включительно)
    - Столбец `critic_score_category` - рейтинг по данным оценок критиков (Высокая оценка, Средняя оценка, Низкая оценка, Нет оценки)
    - Столбец `user_score_category` - рейтинг по данным оценок пользователей (Высокая оценка, Средняя оценка, Низкая оценка, Нет оценки)
    - Столбец `total_sales` - суммарные продажи всех рынков

Новый датафрейм games_df_actual содержит 14 колонок и 12 780 строк